app/src/main/java/swati4star/createpdf/fragment/ZipToPdfFragment.java

```diff
@@ -0,0 +1,160 @@
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.dd.morphingbutton.MorphingButton;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import swati4star.createpdf.R;
+import swati4star.createpdf.activity.MainActivity;
+import swati4star.createpdf.util.Constants;
+import swati4star.createpdf.util.FileUtils;
+import swati4star.createpdf.util.RealPathUtil;
+import swati4star.createpdf.util.StringUtils;
+
+import static android.app.Activity.RESULT_OK;
+
+public class ZipToPdfFragment extends Fragment {
+    private static final int INTENT_REQUEST_PICKFILE_CODE = 10;
+    private String mPath;
+    private Activity mActivity;
+    private FileUtils mFileUtils;
+
+    @BindView(R.id.selectFile)
+    MorphingButton selectFileButton;
+    @BindView(R.id.zip_to_pdf)
+    MorphingButton convertButton;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        View rootview = inflater.inflate(R.layout.fragment_zip_to_pdf, container, false);
+        ButterKnife.bind(this, rootview);
+        mActivity = getActivity();
+        mFileUtils = new FileUtils(mActivity);
+        return rootview;
+    }
+
+    @OnClick(R.id.selectFile)
+    public void showFileChooser() {
+        String folderPath = Environment.getExternalStorageDirectory() + "/";
+        Intent intent = new Intent();
+        intent.setAction(Intent.ACTION_GET_CONTENT);
+        Uri myUri = Uri.parse(folderPath);
+        intent.setDataAndType(myUri, "application/zip");
+
+        startActivityForResult(Intent.createChooser(intent, getString(R.string.merge_file_select)),
+                INTENT_REQUEST_PICKFILE_CODE);
+    }
+
+    public void onActivityResult(int requestCode, int resultCode, Intent data) throws NullPointerException {
+        if (data == null || resultCode != RESULT_OK || data.getData() == null)
+            return;
+        if (requestCode == INTENT_REQUEST_PICKFILE_CODE) {
+            mPath = RealPathUtil.getRealPath(getContext(), data.getData());
+            if (mPath != null) {
+                convertButton.setVisibility(View.VISIBLE);
+            }
+        }
+    }
+
+    @OnClick(R.id.zip_to_pdf)
+    public void convertZipToPdf() {
+        final int BUFFER_SIZE = 4096;
+
+        Snackbar extractingSnackbar = StringUtils.showIndefiniteSnackbar(mActivity,
+                getString(R.string.extracting_images_message));
+        extractingSnackbar.show();
```

Comment:
Have you considered showing a progress bar instead of a snackbar?

Reply:
How will I measure progress then?
Directly going from 0 to 100% won't look good

Reply:
Are you showing progress in the snackbar? I couldn't find out the code where you're doing that. Can you point me to it?

Reply:
I am only showing snackbar untill all the files are extracted or a error occurs.

Reply:
Yes, so when the files are getting extracted, I am suggesting you to show a progress dialog rather than a snackbar. Like this: ![](https://i.stack.imgur.com/pKXqG.jpg)

Reply:
okay..will do it